src/solidlsp/language_servers/nixd_ls.py

```diff
@@ -28,6 +29,82 @@ class NixLanguageServer(SolidLanguageServer):
     Provides Nix specific instantiation of the LanguageServer class using nixd.
     """
 
+    def _extend_nix_symbol_range_to_include_semicolon(
+        self, symbol: ls_types.UnifiedSymbolInformation, file_content: str
+    ) -> ls_types.UnifiedSymbolInformation:
+        """
+        Extend symbol range to include trailing semicolon for Nix attribute symbols.
+
+        nixd provides ranges that exclude semicolons (expression-level), but serena needs
+        statement-level ranges that include semicolons for proper replacement.
+        """
+        if not symbol.get("range"):
+            return symbol
+
+        range_info = symbol["range"]
+        end_line = range_info["end"]["line"]
+        end_char = range_info["end"]["character"]
+
+        # Split file content into lines
+        lines = file_content.split("\n")
+        if end_line >= len(lines):
+            return symbol
+
+        line = lines[end_line]
+
+        # Check if there's a semicolon immediately after the current range end
+        if end_char < len(line) and line[end_char] == ";":
```

Comment:
is the first condition not always true?

Reply:
Maybe not. If the code being replaced it malformed, i.e. no trailing semicolon then the end would be the eol. I'm not sure if the selection would always require a trailing semicolon. If so, then yes, then the first check is redundant.